pep-0646.rst

```diff
@@ -850,65 +850,81 @@ However, note that slices involving starred expressions are still invalid:
 ``*args`` as a TypeVarTuple
 ---------------------------
 
-The second change enables use of ``*args: *Ts``.
+The second change enables use of ``*args: *Ts`` in function definitions.
 
 Before:
 
 ::
 
-    star_etc[StarEtc*]:
-    | '*' a=param_no_default b=param_maybe_default* c=[kwds] {
-        _PyPegen_star_etc(p, a, b, c) }
-    | '*' ',' b=param_maybe_default+ c=[kwds] {
-        _PyPegen_star_etc(p, NULL, b, c) }
-    | a=kwds { _PyPegen_star_etc(p, NULL, NULL, a) }
+    star_etc:
+    | '*' param_no_default param_maybe_default* [kwds]
+    | '*' ',' param_maybe_default+ [kwds]
+    | kwds
 
 After:
 
 ::
 
-    star_etc[StarEtc*]:
-    | '*' a=param_no_default b=param_maybe_default* c=[kwds] {
-        _PyPegen_star_etc(p, a, b, c) }
-    | '*' a=param_no_default_star_annotation b=param_maybe_default* c=[kwds] {
-        _PyPegen_star_etc(p, a, b, c) }
-    | '*' ',' b=param_maybe_default+ c=[kwds] {
-        _PyPegen_star_etc(p, NULL, b, c) }
-    | a=kwds { _PyPegen_star_etc(p, NULL, NULL, a) }
+    star_etc:
+    | '*' param_no_default param_maybe_default* [kwds]
+    | '*' param_no_default_star_annotation param_maybe_default* [kwds]
+    | '*' ',' param_maybe_default+ [kwds]
+    | kwds
 
 Where:
 
 ::
 
-    param_no_default_star_annotation[arg_ty]:
-    | a=param_star_annotation ',' tc=TYPE_COMMENT? { _PyPegen_add_type_comment_to_arg(p, a, tc) }
-    | a=param_star_annotation tc=TYPE_COMMENT? &')' { _PyPegen_add_type_comment_to_arg(p, a, tc) }
+    param_no_default_star_annotation:
+    | param_star_annotation ',' TYPE_COMMENT?
+    | param_star_annotation TYPE_COMMENT? &')'
 
-    param_star_annotation[arg_ty]: a=NAME b=star_annotation { _PyAST_arg(a->v.Name.id, b, NULL, EXTRA) }
+    param_star_annotation: NAME star_annotation
 
-    star_annotation[expr_ty]: ':' a=star_expression { _PyAST_Tuple(_PyPegen_singleton_seq(p, a), Load, EXTRA) }
+    star_annotation: ':' star_expression
 
-This accomplishes the desired outcome (making ``*args: *Ts`` not be a syntax error)
-while matching the behaviour of a starred TypeVarTuple in an index: at runtime,
-``__iter__`` is called on the TypeVarTuple, and a tuple containing the resulting
-unpacked version of the TypeVarTuple is set as the type annotation for ``args``:
+This accomplishes the desired outcome (making ``*args: *Ts`` not be a syntax
+error) while matching the behaviour of star-unpacking in other contexts:
+at runtime, ``__iter__`` is called on the starred object, and a tuple
+containing the items of the resulting iterator is set as the type annotion
+for ``args``. In other words, ``*args: *foo`` is equivalent to
+``*args: tuple(foo)``.
```

Comment:
I'd add "at runtime" somewhere in this last sentence, to make it absolutely clear that this is not an equivalence for static type checkers.

Reply:
Done.